modules/markup/html.go

```diff
@@ -377,8 +377,17 @@ func (ctx *postProcessCtx) visitNode(node *html.Node, visitText bool) {
 
 					lnk := string(link)
 					lnk = util.URLJoin(prefix, lnk)
+
 					link = []byte(lnk)
 				}
+				if setting.CamoServerURL != "" {
```

Comment:
You may also want to ensure `settings.CamoHMACKey` is not empty as well, to avoid an accidental insecure configuration (since the default value defined [here][1] appears to be empty string).

[1]: https://github.com/go-gitea/gitea/pull/12802/files#diff-55df4b4479a1b6872019881b784cf0166b93cd67ce5067b8a04c8b2646518d14R751